src/helper/remote_config/product.cpp

```diff
@@ -44,33 +44,43 @@ void dds::remote_config::product::assign_configs(
     const std::unordered_map<std::string, config> &configs)
 {
     std::unordered_map<std::string, config> to_update;
-    // determine what each config given is
-    for (const auto &[name, config] : configs) {
-        auto previous_config = configs_.find(name);
-        if (previous_config == configs_.end()) { // New config
-            auto config_to_update = config;
-            config_to_update.apply_state = dds::remote_config::protocol::
-                config_state::applied_state::UNACKNOWLEDGED;
-            to_update.emplace(name, config_to_update);
-        } else { // Already existed
-            if (config.hashes ==
-                previous_config->second.hashes) { // No changes in config
-                to_update.emplace(name, previous_config->second);
-            } else { // Config updated
+    bool changes = false;
+
+    if (configs.empty()) {
+        unapply_configs(configs_);
+    } else {
+        // determine what each config given is
+        for (const auto &[name, config] : configs) {
+            auto previous_config = configs_.find(name);
+            if (previous_config == configs_.end()) { // New config
+                changes = true;
                 auto config_to_update = config;
                 config_to_update.apply_state = dds::remote_config::protocol::
                     config_state::applied_state::UNACKNOWLEDGED;
                 to_update.emplace(name, config_to_update);
+            } else { // Already existed
+                if (config.hashes ==
+                    previous_config->second.hashes) { // No changes in config
+                    to_update.emplace(name, previous_config->second);
+                } else { // Config updated
+                    changes = true;
+                    auto config_to_update = config;
+                    config_to_update.apply_state = dds::remote_config::
+                        protocol::config_state::applied_state::UNACKNOWLEDGED;
+                    to_update.emplace(name, config_to_update);
+                }
+                // configs_ at the end of this loop will contain only configs
+                // which have to be unapply. This one has been classified as
+                // something else and therefore, it has to be removed
+                configs_.erase(previous_config);
             }
-            // configs_ at the end of this loop will contain only configs
-            // which have to be unapply. This one has been classified as
-            // something else and therefore, it has to be removed
-            configs_.erase(previous_config);
         }
-    }
 
-    update_configs(to_update);
-    unapply_configs(configs_);
+        if (changes) {
```

Comment:
Shouldn't you also check if `configs_` is not empty?

Reply:
Which scenario do you have in mind?

Reply:
What if one config is removed but no config is changed?